src/Core/src/Response.php

```diff
@@ -85,24 +86,74 @@ public function resolve(?float $timeout = null): bool
                 }
             }
 
-            $statusCode = $this->httpResponse->getStatusCode();
+            return $this->handleStatus();
         } catch (TransportExceptionInterface $e) {
             throw $this->resolveResult = new NetworkException('Could not contact remote server.', 0, $e);
         }
+    }
 
-        if (500 <= $statusCode) {
-            throw $this->resolveResult = new ServerException($this->httpResponse);
-        }
+    /**
+     * Make sure all provided requests are executed.
+     *
+     * @param self[]     $responses
+     * @param float|null $timeout   Duration in seconds before aborting. When null wait
+     *                              until the end of execution. Using 0 means non-blocking
+     *
+     * @return iterable<self>
+     *
+     * @throws NetworkException
+     * @throws HttpException
+     */
+    final public static function multiplex(iterable $responses, float $timeout = null): iterable
+    {
+        /** @var self[] $responseMap */
+        $responseMap = [];
+        $httpResponses = [];
+        $httpClient = null;
+        foreach ($responses as $response) {
+            if (null !== $response->resolveResult) {
+                yield $response;
+
+                continue;
+            }
 
-        if (400 <= $statusCode) {
-            throw $this->resolveResult = new ClientException($this->httpResponse);
+            if (null === $httpClient) {
+                $httpClient = $response->httpClient;
+            }
+            $httpResponses[] = $response->httpResponse;
+            $responseMap[\spl_object_id($response->httpResponse)] = $response;
         }
 
-        if (300 <= $statusCode) {
-            throw $this->resolveResult = new RedirectionException($this->httpResponse);
+        // response is empty
+        if (null === $httpClient) {
+            return;
         }
 
-        return $this->resolveResult = true;
+        try {
+            foreach ($httpClient->stream($httpResponses, $timeout) as $httpResponse => $chunk) {
+                if ($chunk->isTimeout()) {
+                    continue;
+                }
+                if ($chunk->isFirst()) {
+                    $response = $responseMap[\spl_object_id($httpResponse)] ?? null;
+
+                    if (null !== $response) {
+                        try {
+                            $response->handleStatus();
+                        } catch (Exception $e) {
+                        }
+                        yield $response;
```

Comment:
`$response->resolveResult instanceof NetworkException|HttpException`, right?

Reply:
the purpose of this `multiplex` method is yielding each time a response is `resolved` (success or failure)

Do you suggest: `yield $response => $isSuccessfull`:
```php
foreach (Result::multiplex($results) as $result => $success) {
  if (!$success) {
    // deal with error
  }
}
```

could also be `yield $response => $exceptionOrNull`

WDYT?

Reply:
No, sorry. 
I just want to confirm the state of the $response. I use my words: 

Is it true, that on this line. The `$response->resolveResult` will always be an instance of `NetworkException` or `HttpException`?

Reply:
oh yes, you're right

Reply:
euhh no... it's either `true|NetworkException|HttpException`

Reply:
You are correct. 
It cannot be null.